Services/Diagnostics/DiagnosticsLogger.cs

```diff
@@ -0,0 +1,128 @@
+            [CallerLineNumber] int lineNumber = 0);        
+    }
+
+    public class DiagnosticsLogger : IDiagnosticsLogger
+    {
+        public struct JsonStruct
+        {
+            public string EventType;
+            public Dictionary<string, object> EventProperties;
+
+            public JsonStruct(string eventType, Dictionary<string, object> eventProps)
+            {
+                this.EventType = eventType;
+                this.EventProperties = eventProps;
+            }
+        }
+
+        private readonly IHttpClient httpClient;
+        private readonly IServicesConfig servicesConfig;
+        private readonly string diagnosticsEndpoint = string.Empty;
+
+        private const string SERVICE_ERROR_EVENT = "ServiceError";
+        private const string SERVICE_START_EVENT = "ServiceStart";
+        private const string SERVICE_HEARTBEAT_EVENT = "ServiceHeartbeat";
+        
+        public DiagnosticsLogger(IHttpClient httpClient, IServicesConfig servicesConfig)
+        {
+            this.httpClient = httpClient;
+            this.servicesConfig = servicesConfig;
+            this.diagnosticsEndpoint = this.servicesConfig.DiagnosticsEndpointUrl + "/diagnosticsevents";
+        }
+
+        public async Task<IHttpResponse> LogServiceStartAsync(string message)
+        {
+            JsonStruct jsonStruct = new JsonStruct(SERVICE_START_EVENT + message, null);
+            return await this.httpClient.PostAsync(this.PrepareRequest(this.diagnosticsEndpoint, jsonStruct));
+        }
+
+        public async Task<IHttpResponse> LogServiceHeartbeatAsync()
+        {
+            JsonStruct jsonStruct = new JsonStruct(SERVICE_HEARTBEAT_EVENT,null);
+            return await this.httpClient.PostAsync(this.PrepareRequest(this.diagnosticsEndpoint, jsonStruct));
+        }
+
+        public async Task<IHttpResponse> LogServiceErrorAsync(
+            string message,
+            string exceptionMessage = "",
+            [CallerMemberName] string callerName = "",
+            [CallerFilePath] string filePath = "",
+            [CallerLineNumber] int lineNumber = 0)
+        {
+            JsonStruct jsonStruct = this.ConvertToJson(message, exceptionMessage, null, callerName, filePath, lineNumber);
+            return await this.httpClient.PostAsync(this.PrepareRequest(this.diagnosticsEndpoint, jsonStruct));
+        }
+
+        public async Task<IHttpResponse> LogServiceErrorAsync(
+            string message,
+            object data,
+            [CallerMemberName] string callerName = "",
+            [CallerFilePath] string filePath = "",
+            [CallerLineNumber] int lineNumber = 0)
+        {
+            JsonStruct jsonStruct = this.ConvertToJson(message, "", data, callerName, filePath, lineNumber);
+            return await this.httpClient.PostAsync(this.PrepareRequest(this.diagnosticsEndpoint, jsonStruct));
+        }
+
+        private JsonStruct ConvertToJson(string message,
```

Comment:
nit: rename to `ConvertServiceErrorToJson` (because of line 117, this method cannot be reused to serialize other types of event)